plugin/scm.go

```diff
@@ -57,7 +57,7 @@ func (p *Plugin) getScmChanges(ctx context.Context, req *request) ([]string, err
 		after := req.Build.After
 
 		// check for branch pr
-		if commitBranch != repoBranch {
+		if strings.HasPrefix(req.Build.Ref, "refs/pull/") && commitBranch != repoBranch {
```

Comment:
I suspect you may be right that this line is the issue for #17

However, the condition `strings.HasPrefix` is already used here; 

https://github.com/bitsbeats/drone-tree-config/blob/3667ad1bc4bae63d6c51e013b0d226527876ec82/plugin/scm.go#L41 

.. when it evaluates to true, the updated line of code would not be considered. It is in the `else` block to the proceeding `else if`, above

Reply:
@mach6 you are right, this is a check before and the change is in the `else`. The comment for the line changed says *check for branch pr*. I wonder if lines 59-62 could be removed?